test/e2e/tests/shield/shield-subscription-management.spec.ts

```diff
@@ -0,0 +1,296 @@
+      )
+      .always()
+      .thenJson(200, SHIELD_ELIGIBILITY_DATA),
+
+    // Mock user events
+    await mockServer
+      .forPost('https://subscription.dev-api.cx.metamask.io/v1/user-events')
+      .thenJson(200, SHIELD_USER_EVENTS_RESPONSE),
+
+    // Mock card subscription creation endpoint
+    await mockServer
+      .forPost(
+        'https://subscription.dev-api.cx.metamask.io/v1/subscriptions/card',
+      )
+      .thenJson(200, {
+        checkoutSessionUrl: MOCK_CHECKOUT_SESSION_URL,
+      }),
+
+    // Mock claims submission endpoint
+    await mockServer
+      .forPost(/https:\/\/.*\.cx\.metamask\.io\/claims/u)
+      .thenJson(200, SHIELD_CLAIMS_RESPONSE),
+
+    // Mock GET claims endpoint - returns list of claims
+    // Returns a mock claim after submission, empty array otherwise
+    await mockServer
+      .forGet(/https:\/\/.*\.cx\.metamask\.io\/claims/u)
+      .thenJson(200, [
+        {
+          id: SHIELD_CLAIMS_RESPONSE.claimId,
+          shortId: 1,
+          createdAt: new Date().toISOString(),
+          updatedAt: new Date().toISOString(),
+          chainId: '1',
+          email: claimData.email,
+          impactedWalletAddress: '0x5cfe73b6021e818b776b421b1c4db2474086a7e1',
+          impactedTxHash: claimData.impactedTxHash,
+          reimbursementWalletAddress: claimData.reimbursementWalletAddress,
+          description: claimData.description,
+          attachments: [],
+          intercomId: `intercom_${SHIELD_CLAIMS_RESPONSE.claimId}`,
+          status: 'created',
+        },
+      ]),
+
+    // Mock cancel subscription endpoint
+    await mockServer
+      .forPost(
+        /https:\/\/subscription\.dev-api\.cx\.metamask\.io\/v1\/subscriptions\/[^/]+\/cancel$/u,
+      )
+      .thenCallback(() => {
+        cancelAtPeriodEnd = true;
+        return {
+          statusCode: 200,
+          json: { ...BASE_SHIELD_SUBSCRIPTION, cancelAtPeriodEnd },
+        };
+      }),
+
+    // Mock renew subscription endpoint
+    await mockServer
+      .forPost(
+        /https:\/\/subscription\.dev-api\.cx\.metamask\.io\/v1\/subscriptions\/[^/]+\/uncancel$/u,
+      )
+      .thenCallback(() => {
+        cancelAtPeriodEnd = false;
+        return {
+          statusCode: 200,
+          json: { ...BASE_SHIELD_SUBSCRIPTION, cancelAtPeriodEnd },
+        };
+      }),
+  ];
+}
+
+describe('Shield Plan Stripe Integration', function () {
+  // TODO: This test is skipped until the claim form is implemented with the final design.
+  // eslint-disable-next-line mocha/no-skipped-tests
+  it.skip('should successfully fill and submit shield claim form', async function () {
+    // Generate test data before the test runs so it can be reused in mocks
+    const testData = {
+      email: generateRandomEmail(),
+      impactedTxHash: generateRandomTxHash(),
+      reimbursementWalletAddress: generateRandomWalletAddress(),
+      description: generateRandomDescription(),
```

Comment:
Using this functions in a test is problematic as this will cause that the test is not deterministic anymore. In our e2e we want stability and for that we need that tests have deterministic results (no live requests which can change responses, or no randomness if it can be avoided).

We could think on another type of test, like fuzzing, to discover new bugs, but I don't think this should be run on PRs rather on a separate job independent of PRs.